force-app/main/default/classes/PMT_RefundService_TEST.cls

```diff
@@ -90,14 +93,106 @@ public with sharing class PMT_RefundService_TEST {
         UTIL_Http_TEST.CalloutMock calloutMock = UTIL_Http_TEST.mockDataUnprocessableResponse();
         PMT_RefundService refundService = new PMT_RefundService();
         Test.startTest();
-        UTIL_Http.Response response = refundService.processRefund('random-id');
+        UTIL_Http.Response response = refundService.processElevateRefund('random-id');
         Test.stopTest();
 
         System.assertEquals(1, calloutMock.getCalloutCounter(),
             '1 callout should be made when the refund callout is made');
         System.assertEquals(UTIL_Http.STATUS_CODE_UNPROCESSABLE_ENTITY, response.statusCode,
-            'The status code should return 404 not found when the Elevate record cannot be found');
+            'The status code should return 422 unprocessable when the transaction is not completed');
         System.assertEquals(System.Label.pmtOriginalPaymentNotCompleted, response.getErrorMessages(),
             'The response body should be convert to original transaction is not completed error message');
     }
+
+    @isTest
+    private static void verifyAPaymentCanBeFullyRefunded() {
```

Comment:
Can we mock the `paymentSelector` to return mock original payments? We do not really need to insert these records to test that the corresponding refund records are created correctly and the original payments and parent opp records are updated correctly, by persisting data, we're essentially testing trigger logic that runs based on these DML actions